### ISSUE TITLE
Issue with declaring a nested type contraint

### ISSUE THREAD

**Issue by MOZGIII** [jongiddy/typle#1]
```rust
pub trait Verifier {
    type Error;

    /// Verify the attestation report.
    fn verify(&self, report: &AttestationReport) -> Result<(), Self::Error>;
}

#[typle(Tuple for 1..=12)]
impl<T> Verifier for T
where
    T: Tuple,
    T<_>: Verifier,
    <T<_> as Verifier>::Error: Into<Box<dyn std::error::Error>>,
{
  // ...
}
```

The above doesn't work: `<T<_> as Verifier>::Error: Into<Box<dyn std::error::Error>>,` specifically

**Comment by jongiddy** [comment]
Thanks for reporting the issue. I will take a look at adding support for this.

In the meantime you can specify the `Error` bound using:
```rust
typle_bound!(i in ..T::LEN => <T<{i}> as Verifier>::Error): Into<Box<dyn std::error::Error>>,
```

**Comment by jongiddy** [comment]
Fixed in `typle v0.9.7`
